Movies/MoviesMVP/Core/CoreData/Repository.swift

```diff
@@ -0,0 +1,102 @@
+// Repository.swift
+// Copyright © RM. All rights reserved.
+
+import CoreData
+import UIKit
+
+protocol DataBaseFilmsProtocol {
+    func getFilms() -> [Films]
+    func addFilms(films: [ResultsFilm])
+    func deleteAllData()
+}
+
+protocol DataBaseDetailsProtocol {
+    func getDetails() -> [Details]
+    func addDetails(details: FilmDescription)
+    func deleteAllDetailsData()
+}
+
+final class CoreDataFilms: DataBaseFilmsProtocol {
+
+    //MARK: - Internal property
+
+    let coreData = CoreDataService.shared
+
+    //MARK: - Internal methods
+
+    func getFilms() -> [Films] {
+        let entityName = String(describing: Films.self)
+        let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: entityName)
+        let fetchObjects = try? coreData.context.fetch(fetchRequest) as? [Films]
+        return fetchObjects ?? []
+    }
+
+    func addFilms(films: [ResultsFilm]) {
+        for element in films {
+            let movie = Films(context: coreData.context)
+            movie.id = element.id
+            movie.overview = element.overview
+            movie.posterPath = element.posterPath
+            movie.title = element.title
+            movie.voteAverage = element.voteAverage
+            coreData.saveContext()
+        }
+    }
+
+    func deleteAllData() {
+        let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: String(describing: Films.self))
+        fetchRequest.returnsObjectsAsFaults = false
+        do {
+            let results = try CoreDataService.shared.context.fetch(fetchRequest)
+            for object in results {
+                guard let objectData = object as? NSManagedObject else { continue }
+                CoreDataService.shared.context.delete(objectData)
+                coreData.saveContext()
+            }
+        } catch {
+            print("Detele all data in \(String(describing: Films.self)) error :", error)
+        }
+    }
+}
+
+final class CoreDataDetails: DataBaseDetailsProtocol {
```

Comment:
```suggestion
final class CoreDataDetails: DataBaseDetailsProtocol {
```
дублировать код не хорошо 2 класса ради 2-3 х изменений подумай как обьеденить